frame/bags-list/src/migrations.rs

```diff
@@ -122,9 +119,10 @@ impl<T: crate::Config<I>, I: 'static> OnRuntimeUpgrade for AddScore<T, I> {
 	}
 
 	#[cfg(feature = "try-runtime")]
-	fn post_upgrade() -> Result<(), &'static str> {
-		let node_count_before = old::TempStorage::<T, I>::take();
-		// Now, the list node data is not corrupt anymore.
+	fn post_upgrade(node_count_before: Vec<u8>) -> Result<(), &'static str> {
+		let node_count_before: u32 = Decode::decode(&mut node_count_before.as_slice())
+			.expect("the state parameter should be something that was generated by pre_upgrade");
```

Comment:
Just making sure, panicking in `post_upgrade` here is acceptable, right? Because this function is only used in try runtime tests?

Reply:
Yes, I think, there are many `assert!` in the `try-runtime` code.

Reply:
correct.